File: src/main/java/su/svn/daybook/domain/model/TagLabel.java
/*
 * This file was last modified at 2020.09.01 19:28 by Victor N. Skurikhin.
 * This is free and unencumbered software released into the public domain.
 * For more information, please refer to <http://unlicense.org>
 * TagLabel.java
 * $Id$
 */

package su.svn.daybook.domain.model;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder(builderClassName = "Builder")
@Table("")
public class TagLabel implements Serializable, DBEntry<Long> {
    static final long serialVersionUID = -102L;

    @Id
    private Long id;

    @Getter
    @Setter
    @NotNull
    @Column("tag")
    private String tag;
}

File: src/main/js/redux/tagLabel.js
/*
 * This file was last modified at 2021.03.21 17:13 by Victor N. Skurikhin.
 * This is free and unencumbered software released into the public domain.
 * For more information, please refer to <http://unlicense.org>
 * tagLabel.js
 * $Id$
 */

export const TAG_LABEL_PENDING = 'TAG_LABEL_PENDING';
export const TAG_LABEL_SUCCESS = 'TAG_LABEL_SUCCESS';
export const TAG_LABEL_ERROR = 'TAG_LABEL_ERROR';

export function tagLabelPending() {
    return {
        type: TAG_LABEL_PENDING
    }
}

export function tagLabelSuccess(value) {
    return {
        type: TAG_LABEL_SUCCESS,
        tag: value
    }
}

export function tagLabelError(error) {
    return {
        type: TAG_LABEL_ERROR,
        error: error
    }
}
